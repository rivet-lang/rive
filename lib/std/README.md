# The `std` package

The `std` package contains functions and types that help in the development
of software made with Rivet.
