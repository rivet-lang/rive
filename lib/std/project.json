{
    "name": "std",
    "description": "The Rivet standard library",
    "version": "0.1.0",
    "authors": ["The Rivet Developers"],
    "dependencies": [
        {
            "name": "core"
        }
    ]
}
