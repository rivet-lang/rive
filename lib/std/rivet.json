{
    "name": "std",
    "description": "The Rivet standard library",
    "version": "0.1.0",
    "authors": ["Jose Mendoza", "Other contributors"],
    "dependencies": [
        {
            "name": "core"
        }
    ]
}
