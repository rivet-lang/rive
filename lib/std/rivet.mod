name: "std"
description: "The Rivet standard library"
version: "0.1.0"
authors: ["The Rivet team"]
dependencies: []
