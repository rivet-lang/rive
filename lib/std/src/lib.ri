// Copyright (C) 2023 The Rivet Developers. All rights reserved.
// Use of this source code is governed by an MIT license that can
// be found in the LICENSE file.

public import std/backtrace;
public import std/console;
public import std/dynlib;
public import std/env;
public import std/errors;
public import std/fs;
public import std/math;
public import std/mem;
public import std/process;
public import std/semver;
public import std/strconv;
public import std/strings;
public import std/sys;
public import std/traits;
