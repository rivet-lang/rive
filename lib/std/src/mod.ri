// Copyright (C) 2022 The Rivet Team. All rights reserved.
// Use of this source code is governed by an MIT license
// that can be found in the LICENSE file.

pub import "std/backtrace";
pub import "std/console";
pub import "std/env";
pub import "std/errors";
pub import "std/fs";
pub import "std/math";
pub import "std/mem";
pub import "std/os";
pub import "std/process";
pub import "std/strconv";
pub import "std/strings";
pub import "std/sys";
pub import "std/traits";
