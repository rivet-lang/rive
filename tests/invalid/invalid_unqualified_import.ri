import std/fs.Path; // ok
import std/process.*; // ok
import std/dynlib.; // fail

import std/hash.{ wyhash }; // warning

func main() {}
