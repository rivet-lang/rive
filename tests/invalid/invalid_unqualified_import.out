tests/invalid/invalid_unqualified_import.ri:3:8: error: invalid syntax for unqualified import
    3 | import std/dynlib.; // fail
      |        ^~~~~~~~~~
      = note: expected a list of symbols or `*`
rivet: error: could not compile module `invalid_unqualified_import`, aborting due to previous error